depresolve/depresolve_test.go

```diff
@@ -0,0 +1,45 @@
+// +build nettest
+
+package depresolve
+
+import (
+	"reflect"
+	"testing"
+
+	"sourcegraph.com/sourcegraph/srclib/dep"
+)
+
+// TestResolveCustomImportPath tests the behavior of ResolveCustomImportPath
+// when called on some common Go package import paths.
+func TestResolveCustomImportPath(t *testing.T) {
+
+	testCases := []struct {
+		ImportPath string
+		Result     *dep.ResolvedTarget
+	}{
+		{"k8s.io/kubernetes/pkg/api", &dep.ResolvedTarget{"https://github.com/kubernetes/kubernetes", "github.com/kubernetes/kubernetes/pkg/api", "GoPackage", "", ""}},
+		{"gopkg.in/inconshreveable/log15.v2", &dep.ResolvedTarget{"https://gopkg.in/inconshreveable/log15.v2", "gopkg.in/inconshreveable/log15.v2", "GoPackage", "", ""}},
+		{"azul3d.org/semver.v2", &dep.ResolvedTarget{"https://azul3d.org/semver.v2", "azul3d.org/semver.v2", "GoPackage", "", ""}},
+		{"sourcegraph.com/sourcegraph/srclib/graph", &dep.ResolvedTarget{"https://sourcegraph.com/sourcegraph/srclib", "sourcegraph.com/sourcegraph/srclib/graph", "GoPackage", "", ""}},
+		{"sourcegraph.com/sourcegraph/sourcegraph/app", &dep.ResolvedTarget{"https://sourcegraph.com/sourcegraph/sourcegraph", "sourcegraph.com/sourcegraph/sourcegraph/app", "GoPackage", "", ""}},
+		{"sourcegraph.com/sourcegraph/srclib/graph", &dep.ResolvedTarget{"https://sourcegraph.com/sourcegraph/srclib", "sourcegraph.com/sourcegraph/srclib/graph", "GoPackage", "", ""}},
+		{"google.golang.org/grpc", &dep.ResolvedTarget{"https://github.com/grpc/grpc-go", "github.com/grpc/grpc-go", "GoPackage", "", ""}},
+		{"google.golang.org/appengine", &dep.ResolvedTarget{"https://github.com/golang/appengine", "github.com/golang/appengine", "GoPackage", "", ""}},
+		{"google.golang.org/cloud", &dep.ResolvedTarget{"https://github.com/GoogleCloudPlatform/gcloud-golang", "github.com/GoogleCloudPlatform/gcloud-golang", "GoPackage", "", ""}},
+		{"google.golang.org/api", &dep.ResolvedTarget{"https://github.com/google/google-api-go-client", "github.com/google/google-api-go-client", "GoPackage", "", ""}},
+		{"golang.org/x/net/context", &dep.ResolvedTarget{"https://github.com/golang/net", "github.com/golang/net/context", "GoPackage", "", ""}},
+		{"golang.org/x/crypto/ssh", &dep.ResolvedTarget{"https://github.com/golang/crypto", "github.com/golang/crypto/ssh", "GoPackage", "", ""}},
+		{"gopkg.in/redis.v3/internal/hashtag", &dep.ResolvedTarget{"https://gopkg.in/redis.v3", "gopkg.in/redis.v3/internal/hashtag", "GoPackage", "", ""}},
+		{"github.com/gorilla/mux", &dep.ResolvedTarget{"https://github.com/gorilla/mux", "github.com/gorilla/mux", "GoPackage", "", ""}},
+		{"github.com/aws/aws-sdk-go/aws/request", &dep.ResolvedTarget{"https://github.com/aws/aws-sdk-go", "github.com/aws/aws-sdk-go/aws/request", "GoPackage", "", ""}},
+	}
+	for _, test := range testCases {
+		got, err := ResolveImportPath(test.ImportPath)
```

Comment:
Suggestion. You're testing the external API of this package, so you could use an external test rather than internal.

To do so, make 3 changes:

1. Use `depresolve.ResolveImportPath` here.
2. Add `_test` suffix to package name (only in this file) so it's `package depresolve_test`.
3. Import `sourcegraph.com/sourcegraph/srclib-go/depresolve`, since it's now a separate package.